src/Files.App/Filesystem/RecentItems.cs

```diff
@@ -205,6 +206,49 @@ public Task<bool> UnpinFromRecentFiles(RecentItem item)
 			}));
 		}
 
+		private NotifyCollectionChangedEventArgs GetChangedActionEventArgs(IReadOnlyList<RecentItem> oldItems, IList<RecentItem> newItems)
+		{
+			var intersection = oldItems.Intersect(newItems);
```

Comment:
I'm not sure this works that way you intend it to. Intersect returns the items which are in common between two lists, not the ones which are different.
Also RecentItem did not propertly implement IEquatable so `intersection` was always empty. I've fixed this in 
595ac69.

Reply:
Edit: I've fixed GetChangedActionEventArgs() in c6d3cb5 the way I think it's right. Could you take a look?

Reply:
totally right, i meant set difference not intersection 😅 

your changes look right to me, thanks for taking a lot at this!